src/if_checks.c

```diff
@@ -237,3 +237,40 @@ struct check_result *check_type_required_but_not_used_in_if(__attribute__((unuse
 	free_string_list(types_to_check);
 	return res;
 }
+
+struct check_result *check_if_calls_template(__attribute__((unused)) const struct
+                                             check_data *data,
+                                             const struct
+                                             policy_node *node)
+{
+	const struct policy_node *parent = node->parent;
+	while (parent &&
+	       (parent->flavor == NODE_OPTIONAL_POLICY || parent->flavor == NODE_IFDEF || parent->flavor == NODE_TUNABLE_POLICY)) {
+		parent = parent->parent;
+	}
+	
+	if (!parent) {
+		return NULL;
+	}
+	
+	const struct if_call_data *id = node->data.ic_data;
+	
+	if (parent->flavor != NODE_INTERFACE_DEF && parent->flavor != NODE_TEMP_DEF) {
+		return make_check_result('S',
+					 S_ID_IF_CALLS_TEMPL,
+					 "parent node of %s has unsupported flavor %d",
+					 id->name,
+					 parent->flavor);
+	}
+	
+	if (look_up_in_template_map(id->name) &&
+	    !look_up_in_template_map(parent->data.str)) {
```

Comment:
We could just check the flavor for NODE_TEMP_DEF rather than doing the hash lookup.

Reply:
updated to `parent->flavor == NODE_INTERFACE_DEF && look_up_in_template_map(ifd->name)`